app/src/main/java/me/saket/dank/urlparser/UrlParser.java

```diff
@@ -367,6 +369,22 @@ public static boolean isGooglePlayUrl(String urlHost, String uriPath) {
     return urlHost.endsWith("play.google.com") && uriPath.startsWith("/store");
   }
 
+  public static boolean isYouTubeUrl(String urlHost) {
+    return (urlHost.endsWith("youtube.com") || urlHost.endsWith("youtu.be"));
+  }
+
+  @Nullable
+  public static String videoIdFromYouTubeUrl(String url){
+    String videoId = null;
+    String regex = "http(?:s)?:\\/\\/(?:m.)?(?:www\\.)?youtu(?:\\.be\\/|be\\.com\\/(?:watch\\?(?:feature=youtu.be\\&)?v=|v\\/|embed\\/|user\\/(?:[\\w#]+\\/)+))([^&#?\\n]+)";
```

Comment:
This could be moved to `UrlParserConfig` similar to how it's done with other regexes used in this class.

Reply:
I could move it to `UrlParserConfig` but I would beed an instance of `UrlParserConfig` to be able to access the pattern inside the static `videoIdFromYouTubeUrl`.  Any suggestions on that?

Reply:
Hmm indeed, I didn't notice that this method is static. I don't have good ideas. Probably closest we can get is to add public static pattern to the `UrlParserConfig` and access it without instance...